packages/components/src/Form/Inputs/InputColor/InputColor.tsx

```diff
@@ -0,0 +1,219 @@
+  ) => {
+    const inputID = useID(id)
+    const validationMessage = useFormContext(props)
+    const initialWhite = polarbrightness2hsv(white())
+    const initialColor = getColorFromText(value || defaultValue)
+
+    const [color, setColor] = useState<SimpleHSV>(initialColor)
+    const [inputTextValue, setInputTextValue] = useState(
+      value || defaultValue || ''
+    )
+    const [isFocused, setIsFocused] = useState(false)
+
+    const handleFocus = () => setIsFocused(true)
+    const handleBlur = () => setIsFocused(false)
+    const wrappedOnFocus = useWrapEvent(handleFocus, onFocus)
+    const wrappedOnBlur = useWrapEvent(handleBlur, onBlur)
+
+    useEffect(() => {
+      if (value && value !== inputTextValue) {
+        setColor(str2simpleHsv(value))
+        !isFocused && setInputTextValue(value)
+      }
+    }, [isFocused, value, inputTextValue])
+
+    const callOnChange = (newColor: SimpleHSV | string) => {
+      if (!onChange || !newColor) return
+      onChange(createEventWithHSVValue(newColor))
+    }
+
+    const setColorState = (newColor: SimpleHSV) => {
+      setColor(newColor)
+      newColor && setInputTextValue(simpleHSVtoFormattedColorString(newColor))
+      callOnChange(newColor)
+    }
+
+    const handleColorChange = (hs: HueSaturation) =>
+      setColorState({ ...hs, v: color.v })
+
+    const handleSliderChange = (event: FormEvent<HTMLInputElement>) =>
+      setColorState({
+        ...color,
+        v: Number(event.currentTarget.value) / 100,
+      })
+
+    const handleInputTextChange = (event: FormEvent<HTMLInputElement>) => {
+      const newValue = event.currentTarget.value
+      setInputTextValue(newValue)
+
+      const isValid = isValidColor(newValue)
+      callOnChange(isValid ? newValue : initialWhite)
+      setColor(getColorFromText(event.currentTarget.value))
+    }
+
+    const content = (
+      <PopoverContent display="flex" flexDirection="column">
+        <ColorWheel
+          size={colorWheelSize}
+          hue={color.h}
+          saturation={color.s}
+          value={color.v}
+          onColorChange={handleColorChange}
+        />
+        <LuminositySlider
+          min={0}
+          max={100}
+          step={1}
+          value={color.v * 100}
+          width={colorWheelSize}
+          onChange={handleSliderChange}
+        />
+      </PopoverContent>
+    )
+
+    const { open, popover, ref: triggerRef } = usePopover({ content })
+
+    return (
+      <Flex>
+        <Swatch
+          ref={triggerRef}
+          color={hsv2hex(color)}
+          borderRadius={hideInput ? 'medium' : 'none'}
+          borderTopLeftRadius="medium"
+          borderBottomLeftRadius="medium"
+          border="1px solid"
+          borderRight={hideInput ? undefined : 'none'}
+          disabled={props.disabled}
+          onClick={open}
+        />
+        {!props.disabled && popover}
+        {!hideInput && (
+          <InputText
+            {...omitFieldProps(props)}
```

Comment:
Since this component doesn't extend `FieldProps` you can also remove `omitFieldProps` like this:
```suggestion
            {...props}
```